### ISSUE TITLE
Only the yolo10n_uint8.onnx model works for inference.

### ISSUE THREAD

**Issue by Matthew-CQ** [Geekgineer/YOLOs-CPP#3]
Hi, Geekgineer,

Thank you for your great work! I encountered an error when using the model (though it works only with yolo10n_uint8.onnx model). Is there something wrong with the decoder part or onnx model? I would appreciate your response at your earliest convenience.

yolo11n.onnx 
![yolo11n.onnx](https://github.com/user-attachments/assets/f458e5bf-3138-48d0-9d64-b77d0042fedb)
 
Other models
![Other models](https://github.com/user-attachments/assets/dd435156-6a5f-421e-b57e-64b76b665ac5)

Best Regards,
Matthew

**Comment by Geekgineer** [comment]
Hello Matthew-CQ, you are welcome please ensure you use the correct model with the corresponding header.

```cpp

// Include necessary headers
#include <opencv2/opencv.hpp>
#include <iostream>
#include <string>

#include "YOLO11.hpp" // Ensure YOLO11.hpp or other version is in your include path

int main()
{
    // Configuration parameters
    const std::string labelsPath = "../models/coco.names";       // Path to class labels
    const std::string modelPath  = "../models/yolo11n.onnx";     // Path to YOLO11 model
    const std::string imagePath  = "../data/dogs.jpg";           // Path to input image
    bool isGPU = true;                                           // Set to false for CPU processing

    // Initialize the YOLO11 detector
    YOLO11Detector detector(modelPath, labelsPath, isGPU);

    // Load an image
    cv::Mat image = cv::imread(imagePath);

    // Perform object detection to get bboxs
    std::vector<Detection> detections = detector.detect(image);

    // Draw bounding boxes on the image
    detector.drawBoundingBoxMask(image, detections);

    // Display the annotated image
    cv::imshow("YOLO11 Detections", image);
    cv::waitKey(0); // Wait indefinitely until a key is pressed

    return 0;
}
```

**Comment by Matthew-CQ** [comment]
Hi Geekgineer,

I am sure that I am using the correct model along with the corresponding header. However, the result is the same as with your code after building.

![image](https://github.com/user-attachments/assets/a15e3fe6-e7d3-48ad-87fd-dc2382ad13b5)

**Comment by Geekgineer** [comment]
Hi  Matthew,

Please make sure you have built the project according to the README.md also the libs like OpenCV match the current versions etc. for the below image it worked for me.

![dogs_o](https://github.com/user-attachments/assets/03ffb245-f3c8-4ebb-a6c4-59f5cc9491ab)

![image](https://github.com/user-attachments/assets/bff54513-cf36-45b9-946d-cd4e4d63f64d)

**Comment by a333klm** [comment]
I have also this problem.
I made sure to have opencv 4.5.5 with this adaptation in the `CmakeLists.txt`

```
cmake_minimum_required(VERSION 3.0.0)
project(yolo_ort)

option(ONNXRUNTIME_DIR "Path to built ONNX Runtime directory." STRING)
message(STATUS "ONNXRUNTIME_DIR: ${ONNXRUNTIME_DIR}")

set(OpenCV_DIR "/home/username/repos/opencv-4.5.5/build")
find_package(OpenCV 4.5.5 REQUIRED)
...
```
But the output is identical to @Matthew-CQ 
In `image_interference.cpp` I added this line of code to verify the OpenCV version:
```
    std::cout << "OpenCV Version: " << CV_VERSION << std::endl;
```
It confirms, that `4.5.5` is used.

**Comment by Geekgineer** [comment]
Hello guys, 

in case you confront issues with the models provided in the project you can get your fresh onnx yolov11 models by exporting the original models or your custom ones using  this colab env:

[https://colab.research.google.com/github/ultralytics/ultralytics/]( https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb
)

regards,

**Comment by Matthew-CQ** [comment]
Hi @Geekgineer @a333klm,

I fixed it by using [notebook from @Geekgineer](https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb) to generate a new ONNX model (yolo11n.onnx). After replacing it, I got the correct output."

The steps are shown below:

1. Run Setup → Predict → Export (change format to ONNX)

2. Download the yolo11n.onnx model and replace the old one.

Regards,
